src/types/venues.ts

```diff
@@ -206,6 +206,12 @@ export interface EmbeddableVenue extends BaseVenue {
   iframeOptions?: Record<string, string>;
 }
 
+export interface AuditoriumVenue extends BaseVenue {
+  template: VenueTemplate.audience;
+  videoPosition: string;
+  sections: [];
```

Comment:
Section as a subcollection is far more extendable in the future. It will contain all the section-related data? Off top of my head: name, vip-status, isBlocked, rows?, columns?, chats

Reply:
_nods_ makes sense.

I'm definitely agreeing on the 'subcollection' path here, the 'rooms as an array' that we have currently is a super annoying antipattern IMO; has caused so much pain/workarounds needed. I think my question here was mostly to confirm that this was just a placeholder, and that we weren't actually planning to implement them as an array on this object/document.

Since the concept of 'sections' and 'tables' are pretty much the same (at least IMO), it might be good to have a more generic/common 'name/'term' that can easily represent the concept for both of them (it may be that 'sections' is good enough, but wanted to explicitly call this out and let it process in the back of your mind). I don't have a particular choice in mind at this stage, so open to ideas.

Reply:
@mike-lvov Have you looked at https://github.com/sparkletown/internal-sparkle-issues/issues/565 at all?

In particular:

> ideally I could:
> 
> - Create a map:
> - Place TABLES with configured seating / layouts directly onto the map
> - Allow people to join a table by clicking a seat (like in Music Venue) direct from the map
> - Have a few different table layouts for different purposes

While that issue is OOS here, I want to make sure the architectural choices we are making here won't limit us from doing something like that in future.

I believe it would probably be easy enough/fine to be able to add this 'tables/sections' collection to a 'map venue' (eg. Party Map), though not sure if/how that would 'work out' beyond that.

Currently we can place 'rooms'/'portals' on the map quite easily. (sidenote: since we added 'unclickable rooms', I wonder if a better 'general name' for this concept/feature might end up being 'placeables' or similar (aka: things you can 'place' on a map)), but the current tables/sections features don't really support being 'placed' like this. It feels a bit like a 'mixing of concerns' to make tables/sections 'directly placeable'. So i'm wondering what your thoughts are on how we would neatly/cleanly support this potential requirement.

My initial thought was that it almost feels like the 'request' there would make having tables/sections as their own 'venue' object make sense. But at the same time, it feels a bit 'heavy handed'/convoluted, and again, a 'mixing of concerns'.

Another thought I just had is that, based on the new 'can have a direct URL subroute to a table/section' pattern, perhaps we could still use the same 'underlying venue' with tables/sections, but instead of linking to the 'outer view' directly from the room/portal/placeable on the map, we could link to a specific table/section. That way we could have multiple rooms/portals/placeables 'deep linking' into the same venue to access different tables/sections. If we did handle it this way (which I'm currently thinking might actually be a really clean implementation, but open to your thoughts/opinions on it), then we may also want/need the ability to 'turn off' the main 'outer view' of a venue with sections/tables, so that the 'leave table' button would take them directly back to the map/etc they were deep linked from, rather than to the venue that they've never actually seen before.

The purpose of me asking here is to ensure that the architectural choices we are making now won't needlessly limit us in future; easier to fix up at this early stage than to refactor it all down the line.

Reply:
Captured the above context on:

- https://github.com/sparkletown/internal-sparkle-issues/issues/133#issuecomment-797141928
- https://github.com/sparkletown/internal-sparkle-issues/issues/565#issuecomment-797141093
- https://github.com/sparkletown/internal-sparkle-issues/issues/127#issuecomment-797142630